crates/rnote-engine/src/fileformats/xoppformat.rs

```diff
@@ -473,22 +473,36 @@ impl XmlLoadable for XoppLayer {
 
 impl XmlWritable for XoppLayer {
     fn write_to_xml(&self, w: &mut xmlwriter::XmlWriter) {
-        w.start_element("layer");
-
-        if let Some(name) = self.name.as_ref() {
-            w.write_attribute("name", name.as_str());
-        }
-
-        for stroke in self.strokes.iter() {
-            stroke.write_to_xml(w);
-        }
-        for text in self.texts.iter() {
-            text.write_to_xml(w);
-        }
-        for image in self.images.iter() {
-            image.write_to_xml(w);
-        }
-        w.end_element();
+        // only do something if we are sure the layer is not empty
+        // Fix for #985
+        let is_empty = self.name.as_ref().is_some()
```

Comment:
The logic is actually inverted to the variable name, it describes when the layer is not empty. Why is the name included here, can't there be layers without names? (my code comment on the layer says it's optional but I actually don't know, I don't remember how this works in xournal++)

Reply:
The layer name is still in there, please take it out of the condition because it does not indicate if a layer is empty.